examples/meta/src/lib.rs

```diff
@@ -1,46 +1,80 @@
 #![no_std]
 
-use gstd::prelude::*;
+use codec::{Decode, Encode};
+use gstd::{msg, prelude::*};
 use gstd_meta::{meta, TypeInfo};
 
 // Metatypes for input and output
-#[derive(TypeInfo)]
+#[derive(TypeInfo, Decode)]
 pub struct MessageInitIn {
-    pub currency: String,
     pub amount: u8,
+    pub currency: Vec<u8>,
 }
 
-#[derive(TypeInfo)]
+#[derive(TypeInfo, Encode)]
 pub struct MessageInitOut {
     pub exchange_rate: Result<u8, u8>,
     pub sum: u8,
 }
 
-#[derive(TypeInfo)]
+impl From<MessageInitIn> for MessageInitOut {
+    fn from(other: MessageInitIn) -> Self {
+        let rate = match String::from_utf8(other.currency)
+            .expect("Unable to parse str")
+            .as_ref()
+        {
+            "USD" => Ok(2),
+            "EUR" => Ok(3),
+            _ => Err(1),
+        };
+
+        Self {
+            exchange_rate: rate,
+            sum: rate.unwrap_or(0) * other.amount,
+        }
+    }
+}
+
+#[derive(TypeInfo, Decode)]
 pub struct MessageIn {
     pub id: Id,
 }
 
-#[derive(TypeInfo)]
+#[derive(TypeInfo, Encode)]
 pub struct MessageOut {
-    pub res: Vec<Result<Wallet, String>>,
+    pub res: Option<Wallet>,
+}
+
+impl From<MessageIn> for MessageOut {
+    fn from(other: MessageIn) -> Self {
+        unsafe {
+            for wallet in WALLETS.iter() {
+                if wallet.id.decimal == other.id.decimal {
+                    return Self {
+                        res: Some(wallet.clone()),
+                    };
+                };
+            }
+
+            Self { res: None }
```

Comment:
I'd prefer a functional style, something like this (need to be verified!)

```suggestion
        unsafe {
            let res = WALLETS.iter().find(|w| w.id.decimal == other.id.decimal).map(Clone::clone);

            Self { res }
```